src/orion/benchmark/task/task_wrapper.py

```diff
@@ -0,0 +1,183 @@
+        of the wrapped task.
+    """
+
+    def __init__(self, task: TaskType, max_trials: int = None):
+        # TODO: Should Wrappers pass the `task` as an argument to the super() constructor?
+        # When/How are these objects deserialized?
+        if isinstance(task, dict):
+            if len(task) != 1:
+                raise ValueError(
+                    f"Expected task configuration dict to have only one key (the task name), but "
+                    f"got {task} instead."
+                )
+            task_name, task_config = task.popitem()
+            task = BenchmarkTask(of_type=task_name, **task_config)
+        super().__init__(max_trials=max_trials or task.max_trials)
+        self.task: TaskType = task
+
+    def call(self, x: Any) -> List[Dict]:
+        return self.task.call(x)
+
+    def get_search_space(self) -> Dict[str, str]:
+        return self.task.get_search_space()
+
+    @property
+    def unwrapped(self) -> Union[TaskType, BaseTask]:
+        """ Returns the 'unwrapped' task. """
+        return self.task.unwrapped
+
+    @property
+    def configuration(self) -> Dict[str, Any]:
+        """Return the configuration of the task wrapper (including the wrapped task's configuration).
+        """
+        return {
+            type(self).__qualname__: {
+                "task": self.task.configuration,
+                "max_trials": self.max_trials,
+            }
+        }
+
+
+HParamsType = TypeVar("HParamsType")
+
+
+class FixTaskDimensionsWrapper(TaskWrapper[TaskType]):
+    """ Wrapper around a Task that fixes the values of some of its input dimensions.
+
+    The value returned by `get_search_space` is also modified.
+
+    Parameters
+    ----------
+    task : TaskType
+        A task to wrap.
+    fixed_dims : Dict[str, Any]
+        Dictionary mapping from the name of a dimension of the task space to the value that
+        dimension should be fixed at.
+    max_trials : int, optional
+        Maximum number of trials, by default None
+    """
+
+    def __init__(self, task: TaskType, fixed_dims: Dict[str, Any], max_trials: int = None):
+        super().__init__(task=task, max_trials=max_trials)
+        self.fixed_dims = fixed_dims
+        # The whole space, from the wrapped task.
+        self._full_space: Space = SpaceBuilder().build(self.task.get_search_space())
+
+        for dimension_name in fixed_dims:
+            if dimension_name not in self._full_space:
+                raise ValueError(
+                    f"Can't fix dimension '{dimension_name}' because it isn't in the task's space: "
+                    f"{self._full_space}."
+                )
+
+        # The part of the space that can be modified.
+        self._space: Space = SpaceBuilder().build(self.get_search_space())
+
+    def call(self, x: Any) -> List[Dict]:
+        """ Calls the wrapped task, adding/changing some dimensions in `x` if necessary. """
+        trial: Trial
+        if isinstance(x, np.ndarray):
```

Comment:
I'm copy-pasting a discussion here because we lost trace with the migration.

> @bouthilx bouthilx on 14 Sep 2021 Member
> Same comment here about the support for different types of x.
> 
>  
> @lebrice lebrice on 6 Dec 2021 Author Contributor
> Hmm I'm starting to think that this TaskWrapper doesn't really have anything to do with the rest of this PR.
> This is a wontfix for me atm.
> 
>  
> @bouthilx bouthilx on 16 Dec 2021 Member
> Oh, I didn't realize it is not used anywhere in this PR. I would propose you move this class and associated tests to another PR and we can look at it in more details when we have time for it.
> 
>  
> @lebrice lebrice 15 days ago Author Contributor
> Fixed: Removed this file and test from this PR.

This refers to the discussion [here](https://github.com/Epistimio/orion/pull/632#discussion_r707550853)

For reference I'm copy-pasting the current state of the discussion as well in case we might as well loose trace.

 
> Member
> @bouthilx bouthilx on 13 Sep 2021
> I think we had a discussion on the topic but I cannot find it. Why would it support different formats for x? Aren't they supposed to be suggestions from the algorithms which are all following the same convention?
> 
>  
> Contributor
> Author
> @lebrice lebrice on 21 Sep 2021
> Being able to pass ndarrays and tensors is useful for testing
> the benchmark class calls task(**params)
> I want these tasks to support receiving one of their dataclasses as an input
> Also, there are tests for most of these cases, and I don't see supporting too many types of inputs as a deal breaker, but up to you.
> 
>  
> Member
> @bouthilx bouthilx on 24 Sep 2021
> Testing like for unit-testing? I honestly do not see why we would want to test the method with other types than what it would be supposed to handle. Supporting other types does not prevent us from supporting typical point format but it does look like feature creep which isn't great either.
> 
> When you say their dataclasses, you mean something like an instance of FcNetTaskHParams? Soon enough we will drop the point array format and use trial objects instead. At that point the benchmarks should follow this transition and support Trials as the input of the call methods. The call method of the tasks is intended to be following the same interface as the functions users would pass to experiment_client.workon() so we need to be careful not to create inconsistencies.
> 
>  
> Contributor
> Author
> @lebrice lebrice on 6 Dec 2021
> Fixed: Made the input type for each task into a TypedDict, (dictionary with type annotations), rather than dataclasses, and the call of ProfetTask now only supports a single type of input.
> 
>  
> Contributor
> Author
> @lebrice lebrice on 6 Dec 2021
> Also customized the __call__ so that Benchmark calling task(**hparam_dict) won't cause an issue.
> 
>  
> Member
> @bouthilx bouthilx on 16 Dec 2021
> It would be simpler to define def call(**x) no? It is a bit weird/inconsistent otherwise, because Oríon expects the functions that are optimized to take the parameters as arguments, not a dict of the parameters.
> 
>  
> Contributor
> Author
> @lebrice lebrice 15 days ago
> I'd currently argue that it isn't. I can explain why in our next 1-1 meeting.
> 
>  
> Member
> @bouthilx bouthilx 14 days ago
> Ok. We should write back a summary of our discussion here for traceability.
> 
>  
> Member
> @bouthilx bouthilx 14 days ago
> I'll try to summarize here. @lebrice Please correct me if I say anything wrong of forget something. So the issue is that the **kwargs input would causing issue with typing linters because child class would define a different interface than **kwargs. I reread the code and I am a bit confused because I don't see child class overwritting the call method.
> 
> An alternative I would see that could be more typing-friendly is to rename call to something like execute and then in each subclass write call(with, the, proper, arguments) and convert them into a dict passed to execute. It does not add value in terms of features, but it makes the code a bit more explicit.
> 
> On the same topic, tasks are expected to rewrite call, not __call__.
> 
>  
> Contributor
> Author
> @lebrice lebrice 16 minutes ago
> Ok, fixed in c3a3846